kythe/extractors/openjdk11/extract.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"flag"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"kythe.io/kythe/go/util/flagutil"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	javaCommandVar      = "KYTHE_JAVA_COMMAND"
+	kytheRootVar        = "KYTHE_ROOT_DIRECTORY"
+	kytheVnameVar       = "KYTHE_VNAMES"
+	javaMakeVar         = "JAVA_CMD"
+	runfilesWrapperPath = "io_kythe/kythe/extractors/openjdk11/java-wrapper"
+	runfilesVnamesPath  = "io_kythe/kythe/extractors/openjdk11/vnames.json"
+)
+
+var (
+	wrapperPath  string
+	vnameRules   string
+	errorPattern = regexp.MustCompile("ERROR: extractor failure for module ([^:]*):")
+)
+
+func setupRunfiles() error {
+	if len(os.Getenv("RUNFILES_DIR")) > 0 || len(os.Getenv("RUNFILES_MANIFEST_FILE")) > 0 {
+		return nil
+	}
+	for _, base := range []string{os.Args[0] + ".runfiles", "."} {
```

Comment:
I propose that the following reorganization would be a little more idiomatic:

```go
	for _, base := range []string{os.Args[0] + ".runfiles", "."} {
		root, err := filepath.Abs(base)
		if err != nil {
			continue
		} else if _, err := os.Stat(root); err != nil {
			continue
		}
		os.Setenv("RUNFILES_DIR", root)
		manifest := filepath.Join(root, "MANIFEST")
		if _, err := os.Stat(manifest); err == nil {
			os.Setenv("RUNFILES_MANIFEST_FILE", manifest)
		}
		return nil
	}
```

Reply:
Done.